### PR TITLE
Filter nodes to only try and send files to external IPs

### PR DESCRIPTION
**Pull request by foxish**
Fixes https://github.com/apache-spark-on-k8s/spark/issues/90

### CODE REVIEW

**Comment by iyanuobidele** [comment]
LGTM. You might wanna check the import statements, I think it got rearranged by your IDE.

**Comment by foxish** [comment]
Should be fixed now. Thanks!

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala [hunk 1]
```diff
@@ -745,9 +745,11 @@ private[spark] class Client(
     val nodeUrls = kubernetesClient.nodes.list.getItems.asScala
       .filterNot(node => node.getSpec.getUnschedulable != null &&
         node.getSpec.getUnschedulable)
-      .flatMap(_.getStatus.getAddresses.asScala.map(address => {
+      .flatMap(_.getStatus.getAddresses.asScala)
+      .filter(_.getType == "ExternalIP")
```
Comment (mccheah) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala]: A quick comment here would be nice. Out of curiosity, is there a reference for this constant somewhere?

Comment (foxish) [reply resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala]: https://kubernetes.io/docs/admin/node/#addresses

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala [hunk 2]
```diff
@@ -746,6 +746,9 @@ private[spark] class Client(
       .filterNot(node => node.getSpec.getUnschedulable != null &&
         node.getSpec.getUnschedulable)
       .flatMap(_.getStatus.getAddresses.asScala)
+      // The list contains hostnames, internal and external IP addresses.
+      // we want only external IP addresses in our list
+      // (https://kubernetes.io/docs/admin/node/#addresses)
```
Comment (mccheah) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala]: Actually, why do we not want external host name here?

Comment (foxish) [reply resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala]: On most systems I see, it is the ExternalIP which is accessible. For example, on GKE/GCE, it is not possible to use the hostnames to get access to the cluster by default. If the hostnames were resolvable, they should resolve to the same external IP address in a given cluster, which is why I don't see the point of including both.

**Comment by mccheah** [comment]
Looks great! Will merge this now.

**Pull request merged**